comp/core/autodiscovery/autodiscoveryimpl/autoconfig.go

```diff
@@ -391,23 +390,19 @@ func (ac *AutoConfig) fillFlare(fb flaretypes.FlareBuilder) error {
 	return nil
 }
 
-// Start will listen to the service channels before anything is sent to them
-// Usually, Start and Stop methods should not be in the component interface as it should be handled using Lifecycle hooks.
-// We make exceptions here because we need to disable it at runtime.
-func (ac *AutoConfig) Start() {
-	ac.startOnce.Do(func() {
-		listeners.RegisterListeners(ac.serviceListenerFactories)
-		providers.RegisterProviders(ac.providerCatalog)
-		setupAcErrors()
-		// Start the service listener
-		go ac.serviceListening()
-	})
+// start will listen to the service channels before anything is sent to them
+func (ac *AutoConfig) start() {
```

Comment:
nit: Would it make sense to have not have the `start()` and `stop()` function at all and just have all the code in the Fx `OnStart` and `OnStop` functions?